components/loading-spinner/loading-spinner.js

```diff
@@ -0,0 +1,172 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+
+class LoadingSpinner extends LitElement {
+
+	static get properties() {
+		return {
+			color: { type: String },
+			size: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return css`
+
+			:host {
+				color: var(--d2l-loading-spinner-color, var(--d2l-color-celestine));
+				display: inline-block;
+				text-align: initial;
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+
+			:host .d2l-loading-spinner-wrapper {
+				height: var(--d2l-loading-spinner-size, 50px);
+				margin: auto;
+				overflow: hidden;
+				position: relative;
+				width: var(--d2l-loading-spinner-size, 50px);
+			}
+			:host .d2l-loading-spinner-bg,
+			:host .d2l-loading-spinner-bg-blur {
+				height: 100%;
+				position: absolute;
+				stroke: var(--d2l-loading-spinner-color, var(--d2l-color-celestine));
+				top: 0;
+				width: 100%;
+			}
+			:host .d2l-loading-spinner-bg-stroke {
+				stroke: #ededfa;
+			}
+			:host .d2l-loading-spinner-bg-blur {
```

Comment:
I know this is existing CSS, but are all the `:host` prefixes necessary?

Reply:
Hmm, I will try to tidy these up.